Modules/Quotas/test/Quotas.Domain.Tests/Tests/Tiers/TierTests.cs

```diff
@@ -184,14 +187,50 @@ public void AddQuotaForAllMetricsOnQueuedForDeletion_only_creates_missing_quotas
         addedQuotas.Should().HaveCount(1);
         tier.Quotas.Should().HaveCount(2);
     }
+
+    [Fact]
+    public void Triggers_TierQuotaDefinitionDeletedDomainEvent()
+    {
+        // Arrange
+        var tierId = new TierId("tier-id");
+        var tier = new Tier(tierId, "some-tier-name");
+
+        tier.CreateQuota(MetricKey.NumberOfSentMessages, 5, QuotaPeriod.Month);
+        var tierQuotaDefinitionId = tier.Quotas.First().Id;
+
+        // Act
+        tier.DeleteQuota(tierQuotaDefinitionId);
+
+        // Assert
+        tier.DomainEvents.Should().HaveCount(2);
+        tier.DomainEvents[1].Should().BeOfType<TierQuotaDefinitionDeletedDomainEvent>();
+        var domainEvent = (TierQuotaDefinitionDeletedDomainEvent)tier.DomainEvents[1];
+        domainEvent.TierId.Should().Be(tierId);
+        domainEvent.TierQuotaDefinitionId.Should().Be(tierQuotaDefinitionId);
```

Comment:
You should use our extension method for this.

(to be able to do this, you should clear the domain events at the end of the Arrange block)